src/sublime_aio.py

```diff
@@ -1,20 +1,36 @@
 from __future__ import annotations
 import asyncio
 import atexit
+import functools
 import traceback
 from inspect import iscoroutinefunction
 from threading import Thread
-from typing import TYPE_CHECKING
-
-if TYPE_CHECKING:
-    from asyncio import Task
-    from concurrent.futures import Future
-    from typing import Any, Callable, Coroutine
+from typing import TYPE_CHECKING, overload
 
 import sublime
 import sublime_api
 import sublime_plugin
 
+if TYPE_CHECKING:
+    from collections.abc import Coroutine
+    from concurrent.futures import Future
+    from typing import Any, Callable, ParamSpec, TypeVar, TypeAlias
+
+    P = ParamSpec('P')
+    T = TypeVar('T')
+
+    BlankCoro: TypeAlias = Coroutine[object, None, None]
+    CompletionsReturnVal: TypeAlias = (
+        sublime.CompletionList
+        | tuple[list[sublime.CompletionValue], sublime.AutoCompleteFlags]
+        | list[sublime.CompletionValue]
+        | None
+    )
+
+    EL = TypeVar('EL', bound=sublime_plugin.EventListener)
+    VEL = TypeVar('VEL', bound=sublime_plugin.ViewEventListener)
```

Comment:
As we use those vars in decorator function which works only within `sublime_aio.EventListeners`, shouldn't those vars point to the more specific classes then?

```suggestion
    EL = TypeVar('EL', bound=EventListener)
    VEL = TypeVar('VEL', bound=ViewEventListener)
```

This way we get feedback of misuse...

![grafik](https://github.com/user-attachments/assets/cc0f844e-09bd-445e-a123-7ec8e4dd3c7b)


_Note: I find it odd those to be needed just to avoid pyright not to throw typing errors. I'd expect following `@overload` to also work when directly using `EventListener` types._

Reply:
Good point. Adopted this.